### PR TITLE
build: Added Dotnet SDK ruleset and build props.

### LINKED ISSUE #16: Add botbuilder-dotnet FxCop and StyleCop packages

### PR DESCRIPTION
**Pull request by tracyboehrer**
Fixes #16 

This PR also corrects errors in the code that resulted from the new ruleset.  There are no logic changes.

### CODE REVIEW

**Comment by tracyboehrer** [comment]
@ryanlengel There are a couple build errors I'm not sure how we need to handle.

File: Directory.Build.props [hunk 1]
```diff
@@ -0,0 +1,84 @@
+<Project>
+  
+  <PropertyGroup>
+    <LocalPackageVersion>4.10.0-local</LocalPackageVersion>
```
Comment (ryanlengel) [inline Directory.Build.props:4]: Is this an SDK package version?

Comment (ryanlengel) [reply Directory.Build.props:4]: i.e. does this need to be in sync with newer packages?

File: src/Microsoft.Bot.Core/CoreBot.cs [hunk 2]
```diff
@@ -40,18 +40,18 @@ public class CoreBot : ActivityHandler
...
-            this.defaultLocale = configuration.GetValue<string>("defaultLanguage") ?? "en-us"; ;
-            this.telemetryClient = telemetryClient;
+            this._conversationState = conversationState;
+            this._userState = userState;
+            this._dialogState = conversationState.CreateProperty<DialogState>("DialogState");
+            this._resourceExplorer = resourceExplorer;
+            this._defaultLocale = configuration.GetValue<string>("defaultLanguage") ?? "en-us";
+            this._telemetryClient = telemetryClient;
 
             /*
```
Comment (ryanlengel) [inline src/Microsoft.Bot.Core/CoreBot.cs:50]: Now that I have issues for these I will clean up the TODOs to reference the # and title only in a minor PR

**Review by ryanlengel — APPROVED** [review]

**Pull request merged**